38.md

```diff
@@ -0,0 +1,170 @@
+
+## Kind 141: Update metadata and Add participants
+
+Update an encrypted channel's public metadata including its particpants.
+
+Clients and relays SHOULD handle kind 141 events similar to kind 0 `metadata` events.
+
+Clients SHOULD ignore kind 141s from pubkeys other than the pubkey that created the given channel using the corresponding kind 140 event. In other words, only the creator of channel can send kind 141 events for a given channel; kind 141 for any channel sent by other people, even participants of that channel, are ignored.
+
+Clients SHOULD support basic metadata fields:
+
+- `name` - string - Channel name
+- `about` - string - Channel description
+- `picture` - string - URL of channel picture
+
+Clients MAY add additional metadata fields.
+
+Clients SHOULD use [NIP-10](10.md) marked "e" tags to recommend a relay.
+
+```json
+{
+    "content": "{\"name\": \"Updated Demo Channel\", \"about\": \"Updating a test channel.\", \"picture\": \"https://placekitten.com/201/201\"}",
+    "tags": [["e", <channel_create_event_id> <relay-url>]],
+    ...
+}
+```
+
+Further, the main use case of kind 141 event is to update the participants list. New p tags may be added, and old ones removed with the 141 event for an encrypted channel. Client will internally then manage participants. It should be noted that since the shared-secret is not changeable,  the 'removed' members can continue to read all channel messages.
+
+If new members are added, then its the responsibility of the creator's client to share the shared-secret with them as mentioned in kind 140 section. 
+
+## Kind 142: Send encrypted message to encrypted group channel
+
+
+Clients SHOULD use [NIP-10](10.md) marked "e" tags to recommend a relay and specify whether it is a reply or root message.
+
+Clients SHOULD append [NIP-10](10.md) "p" tags to replies.
+
+### Format
+
+Root message:
+
+```json
+{
+    "content": <string>,
+    "tags": [["e", <kind_140_event_id>],
+    ...
+}
+```
+
+Reply to another message:
+
+```json
+{
+    "content": <string>,
+    "tags": [
+        ["e", <kind_140_event_id>],
+        ["e", <kind_142_event_id_replied_to>], 
+        ["p", <pubkey> <relay-url>],
+        
+        ...
+    ],
+    ...
+}
+```
+
+## Kind 104: Communicate the Shared-Secret
+
+Upon adding a new member to a encrypted group of kind 140, the creator of the channel should send a direct message ( of kind 104) to each of the other participants, which will have the channel's shared-secret private/public keys. The direct message should of `kind` 104; it must follow the message construction rules as are there for `kind 4` messages, except for the difference in value of `kind`.
+
+The format of that message is:
+
+`App Encrypted Channels: inviting you to encrypted channel <channel id> encrypted using private public keys <shared private key> <corresponding shared public key>`
```

Comment:
This message feels easy for humans to read and hard for machines to read. We should optimize for the opposite. Also, the public key can be derived from the public key so it does not need to be included.

My suggestion would be `content="<shared private key>"`

Then, include the <channel id> as a tag of some sort.

Reply:
this should not use a nip04 style encryption, instead it should use directional encryption 

encrypt({
sender pub,
sender sig of this payload,
content=....
tags=...
}, ephemeral pub),

no tags outside the envelope, pubkey is ephemeral.   recipient verifies the sender's provenance after decryption

*all* messages should work this way in this protocol